manim/mobject/functions.py

```diff
@@ -3,22 +3,22 @@
 from ..mobject.types.vectorized_mobject import VMobject
 from ..utils.config_ops import digest_config
 import math
+import attr
+import typing as tp
 
 
+@attr.s(auto_attribs=True, eq=False)
 class ParametricFunction(VMobject):
-    CONFIG = {
-        "t_min": 0,
-        "t_max": 1,
-        "step_size": 0.01,  # Use "auto" (lowercase) for automatic step size
-        "dt": 1e-8,
-        # TODO, be smarter about figuring these out?
-        "discontinuities": [],
-    }
+    t_min: float = 0
+    t_max: float = 1
+    step_size: tp.Union[str, float] = 0.01  # Use "auto" (lowercase) for automatic step size
```

Comment:
Comment: note that, if on python 3.8+, `str` can be replaced by `tp.Literal["auto"]`. I suppose we can add a compatibility check: (before the class)
`auto_lit = tp.Literal["auto"] if hasattr(tp, "Literal") else str`
then use it here, so that people in python 3.8+ can have proper typings
but that's just an idea